src/Umbraco.Infrastructure/HostedServices/RecurringHostedServiceBase.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+
+namespace Umbraco.Infrastructure.HostedServices
+{
+    /// <summary>
+    /// Provides a base class for recurring background tasks implemented as hosted services.
+    /// </summary>
+    /// <remarks>
+    /// See: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services?view=aspnetcore-3.1&tabs=visual-studio#timed-background-tasks
+    /// </remarks>
+    public abstract class RecurringHostedServiceBase : IHostedService, IDisposable
+    {
+        protected const int DefaultDelayMilliseconds = 180000; // 3 mins
+
+        private readonly int _periodMilliseconds;
+        private readonly int _delayMilliseconds;
+        private Timer _timer;
+
+        protected RecurringHostedServiceBase(int periodMilliseconds, int delayMilliseconds)
```

Comment:
If this uses `TimeSpan` as parameters. it becomes much more flexible.

Reply:
Have updated to use `TimeSpan` now.